nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -1341,6 +1348,43 @@ impl DataStore {
         opctx: &OpContext,
         authz_instance: &authz::Instance,
         ok_to_delete_instance_states: &'static [InstanceState],
+    ) -> DeleteResult {
+        // First, mark the instance record as destroyed and detach all disks.
+        //
+        // We do this before other cleanup to prevent concurrent operations from
+        // accessing and modifying the instance while it's being torn down.
+        self.instance_mark_destroyed_and_detach_disks(
+            opctx,
+            authz_instance,
+            ok_to_delete_instance_states,
+        )
+        .await?;
+
+        // Next, delete all other data associated with the instance.
+        //
+        // Note that due to idempotency of this function, it's possible that
+        // "authz_instance.id()" has already been deleted.
+        let instance_id = InstanceUuid::from_untyped_uuid(authz_instance.id());
+        self.instance_ssh_keys_delete(opctx, instance_id).await?;
+        self.instance_mark_migrations_deleted(opctx, instance_id).await?;
+
+        Ok(())
+    }
+
+    // Marks the instance "Destroyed" and detaches disks.
+    //
+    // This is only one internal part of destroying an instance, see:
+    // [`project_delete_instance`] for a more holistic usage.
```

Comment:
nitpicky: perhaps this comment ought to explain why this is factored out? not a big deal.

Reply:
Added a comment saying "it has been refactored out for readability"